packages/kit/src/core/adapt/prerender/prerender.js

```diff
@@ -167,8 +167,16 @@ export async function prerender({ cwd, out, log, config, build_data, fallback, a
 				if (location) {
 					mkdirp(dirname(file));
 
+					let url;
+					try {
+						url = new URL(location);
+					} catch (error) {
+						log.error(`'${location}' is not a valid URL`);
+						url = location;
+					}
+
 					log.warn(`${rendered.status} ${decoded_path} -> ${location}`);
-					writeFileSync(file, `<meta http-equiv="refresh" content="0;url=${encodeURI(location)}">`);
+					writeFileSync(file, `<meta http-equiv="refresh" content="0;url=${url.toString()}">`);
```

Comment:
This should probably just be `location`?
```suggestion
					writeFileSync(file, `<meta http-equiv="refresh" content="0;url=${location}">`);
```

Reply:
I'm not sure why it was originally set to `encodeURI`. I didn't want to break it so I first convert it to an `URL` object.
Is it possible that a unencoded url reaches this part? I think yes, so we can't use `location` here.

Reply:
Spelunking through the git history led me here: https://github.com/sveltejs/kit/pull/256

I think we need to escape the string, so that this doesn't cause havoc...

```
location: "><script>alert('pwned')</script>
```

...but it seems like `encodeURI` was probably the wrong choice.

Reply:
> ```
> location: "><script>alert('pwned')</script>
> ```
This will no longer be a problem because it is an invalid URL. The code will throw an error

Reply:
This is a valid (unencoded) URL:

```
https://example.com/"><script>alert('pwned')</script>
```

Reply:
and with this implementation, this would output the url as:
`"https://example.com/%22%3E%3Cscript%3Ealert('pwned')%3C/script%3E"`